### PR TITLE
fix(frontend): auth logic

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**
1. In saas mode, we would call `authenticate` before any GitHub token was set
2. The oauth redirect uri was incorrect

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
1. If saas mode, ensure authenticate is only called if gh token is present
2. Fix oauth redirect uri

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:6a91b44-nikolaik   --name openhands-app-6a91b44   docker.all-hands.dev/all-hands-ai/openhands:6a91b44
```

### CODE REVIEW

**Review by tofarr — APPROVED** [review]
🍰

File: frontend/src/entry.client.tsx [hunk 1]
```diff
@@ -52,8 +52,8 @@ async function prepareApp() {
 
 const queryClient = new QueryClient({
   queryCache: new QueryCache({
-    onError: (error) => {
-      toast.error(error.message);
+    onError: (error, query) => {
+      if (!query.queryKey.includes("authenticated")) toast.error(error.message);
```
Comment (rbren) [inline frontend/src/entry.client.tsx:56]: this seems like a bit of a hack--is there a better way?

File: frontend/src/utils/generate-github-auth-url.ts [hunk 2]
```diff
@@ -5,6 +5,6 @@
  * @returns The URL to redirect to for GitHub OAuth
  */
 export const generateGitHubAuthUrl = (clientId: string, requestUrl: URL) => {
-  const redirectUri = `${requestUrl.origin}/oauth/github/callback`;
+  const redirectUri = `${requestUrl.origin}/oauth`;
```
Comment (rbren) [inline frontend/src/utils/generate-github-auth-url.ts:8]: this is in our github settings and hard to change without causing an outage. Can we revert back to the old one?

File: frontend/src/entry.client.tsx (hunk 1 above)
Comment (amanape) [reply frontend/src/entry.client.tsx:56]: Previously we could handle them in the individual queries but the latest versions of TanStack Query remove these callbacks from `useQuery`, so AFAIK there is no straight forward way.

The alternative would be rendering the error toasts on failures manually (but does repeat itself).

@tofarr Do you know of any alternatives?

**Comment by rbren** [comment]
Let's get this in for now

**Pull request merged**